src/main/java/de/maxhenkel/pipez/blocks/tileentity/types/ItemPipeType.java

```diff
@@ -136,7 +140,7 @@ protected void insertEqually(PipeLogicTileEntity tileEntity, Direction side, Lis
     protected void insertOrdered(PipeLogicTileEntity tileEntity, Direction side, List<PipeTileEntity.Connection> connections, IItemHandler itemHandler) {
         int itemsToTransfer = getRate(tileEntity, side);
 
-        ArrayList<ItemStack> nonFittingItems = new ArrayList<>();
+        HashSet<Item> nonFittingItems = new HashSet<>();
```

Comment:
Example scenario:
You are inserting an item with NBT data. If that item couldn't be inserted, just the item would get added to the set. If you now want to insert another itemstack that has the same item with different NBT data but would be able to get inserted into the inventory, this code would basically prevent that item from being inserted.

Please correct me if I'm wrong.

Reply:
i will check this, but im pretty sure the `Item` type has NBT data included, so HashSet::contains(differentNBTItem) should return false.

Reply:
No. The Item class is not holding any data. There's only a single instance per-item of any item object.